src/main/java/frc/robot/subsystems/MoverSubsystem.java

```diff
@@ -0,0 +1,130 @@
+    private final double EXTENSION_ROT_TO_M = Units.inchesToMeters(0.5615); 
+
+    private final boolean RESET_OFFSET_ON_STAGE_SWITCH = true;
+    private double angleOffset = 0; //radians
+    private double extensionOffset = 0; //meters
+
+    private MoverPosition currentState = MoverPosition.GROUND;
+
+    public enum GamePiece {
+        CONE, CUBE;
+    }
+    
+    public enum MoverPosition {
+        VERTICAL(0,0),
+        GROUND(Units.degreesToRadians(90), distanceToExtension(25)),
+        SUBSTATION(0,0),
+        CUBEMID(0,0), CUBEHIGH(0,0), 
+        CONEMID(0,0), CONEHIGH(0,0)
+        ;
+
+        public double angle; //Radians
+        public double extension; //Meters, 0 extension is 25 inches from pivot
+
+        private MoverPosition(double angle, double extension){
+            this.angle = angle;
+            this.extension = extension;
+        }
+    }
+
+    public MoverSubsystem(){
+        rotationMotor = MotorUtil.createSparkMax(ROTATION_MOTOR_PORT);
+        rotationMotor.setIdleMode(IdleMode.kBrake);
+
+        rotationEncoder = rotationMotor.getEncoder();
+        rotationEncoder.setPositionConversionFactor(ROTATION_ROT_TO_RAD);
+        
+        rotationMotor.enableSoftLimit(SoftLimitDirection.kForward, true);
+        rotationMotor.enableSoftLimit(SoftLimitDirection.kReverse, true);
+        rotationMotor.setSoftLimit(SoftLimitDirection.kForward, (float) Units.degreesToRadians(90));
+        rotationMotor.setSoftLimit(SoftLimitDirection.kReverse, (float) Units.degreesToRadians(-90));
+
+        rotationPidController = rotationMotor.getPIDController();
+        rotationPidController.setP(ROTATION_P);
+        rotationPidController.setI(ROTATION_I);
+        rotationPidController.setD(ROTATION_D);
+
+
+        extensionMotor = MotorUtil.createSparkMax(EXTENSION_MOTOR_PORT);
+        extensionMotor.setIdleMode(IdleMode.kBrake);
+
+        extensionEncoder = extensionMotor.getEncoder();
+        extensionEncoder.setPositionConversionFactor(EXTENSION_ROT_TO_M);
+        
+        extensionMotor.enableSoftLimit(SoftLimitDirection.kForward, true);
+        extensionMotor.enableSoftLimit(SoftLimitDirection.kReverse, true);
+        extensionMotor.setSoftLimit(SoftLimitDirection.kForward, (float) Units.inchesToMeters(29));
+        extensionMotor.setSoftLimit(SoftLimitDirection.kReverse, (float) Units.inchesToMeters(0));
+
+        extensionPidController = extensionMotor.getPIDController();
+        extensionPidController.setP(EXTENSION_P);
+        extensionPidController.setI(EXTENSION_I);
+        extensionPidController.setD(EXTENSION_D);
+
+    }
+
+    @Override
+    public void periodic(){
+        goTo(currentState.angle + angleOffset, currentState.extension + extensionOffset);
+    }
+
+
+    private void goTo(double angle, double extension){
+        rotationPidController.setReference(angle, ControlType.kPosition);
+        extensionPidController.setReference(extension, ControlType.kPosition);
```

Comment:
it will really help to put data on target/current position on shuffleboard, along with being able to tune the PID constants there. otherwise it will take a long time to wait for reupload every time.